src/shader/ShaderDynamic.js

```diff
@@ -742,19 +742,33 @@ x3dom.shader.DynamicShader.prototype.generateFragmentShader = function ( gl, pro
         }
     }
 
-    //Fog
-    if ( properties.FOG )
-    {
-        shader += x3dom.shader.fog();
-    }
-
-    if ( properties.LIGHTS || properties.CLIPPLANES )
+    // //Fog
+    // if ( properties.FOG )
+    // {
+    //     shader += x3dom.shader.fog();
+    // }
+
+    // if ( properties.LIGHTS || properties.CLIPPLANES )
+    // {
+    //     shader += "varying vec4 fragPosition;\n";
+    //     shader += "varying vec4 fragPositionWS;\n";
+    //     shader += "uniform float isOrthoView;\n";
+    // }
+
+    // same as vertex shader but with fragPositionWS for fogNoise (w/ or w/out lights)
+    if ( properties.LIGHTS || properties.FOG || properties.CLIPPLANES )
     {
+        shader += "uniform vec3 eyePosition;\n";
```

Comment:
Is this needed ?

Reply:
looks like it is only needed by the vertex shader, everything seems to work w/out it.